src/algebra/order/ring.lean

```diff
@@ -102,285 +115,290 @@ lemma add_one_le_two_mul [has_le α] [semiring α] [covariant_class α α (+) (
 calc  a + 1 ≤ a + a : add_le_add_left a1 a
         ... = 2 * a : (two_mul _).symm
 
-/-- An `ordered_semiring α` is a semiring `α` with a partial order such that
-addition is monotone and multiplication by a positive number is strictly monotone. -/
+/-- An `ordered_semiring α` is a semiring with a partial order such that addition is monotone and
+multiplication by a positive number is strictly monotone. -/
 @[protect_proj]
-class ordered_semiring (α : Type u) extends semiring α, ordered_cancel_add_comm_monoid α :=
+class ordered_semiring (α : Type u) extends semiring α, ordered_add_comm_monoid α :=
 (zero_le_one : (0 : α) ≤ 1)
-(mul_lt_mul_of_pos_left  : ∀ a b c : α, a < b → 0 < c → c * a < c * b)
-(mul_lt_mul_of_pos_right : ∀ a b c : α, a < b → 0 < c → a * c < b * c)
+(mul_le_mul_of_nonneg_left  : ∀ a b c : α, a ≤ b → 0 ≤ c → c * a ≤ c * b)
+(mul_le_mul_of_nonneg_right : ∀ a b c : α, a ≤ b → 0 ≤ c → a * c ≤ b * c)
 
 section ordered_semiring
 variables [ordered_semiring α] {a b c d : α}
 
-lemma mul_lt_mul_of_pos_left (h₁ : a < b) (h₂ : 0 < c) : c * a < c * b :=
-ordered_semiring.mul_lt_mul_of_pos_left a b c h₁ h₂
+lemma mul_le_mul_of_nonneg_left : a ≤ b → 0 ≤ c → c * a ≤ c * b :=
+ordered_semiring.mul_le_mul_of_nonneg_left _ _ _
 
-lemma mul_lt_mul_of_pos_right (h₁ : a < b) (h₂ : 0 < c) : a * c < b * c :=
-ordered_semiring.mul_lt_mul_of_pos_right a b c h₁ h₂
+lemma mul_le_mul_of_nonneg_right : a ≤ b → 0 ≤ c → a * c ≤ b * c :=
+ordered_semiring.mul_le_mul_of_nonneg_right _ _ _
 
 @[priority 100] -- see Note [lower instance priority]
 instance ordered_semiring.zero_le_one_class : zero_le_one_class α :=
 { ..‹ordered_semiring α› }
 
 @[priority 200] -- see Note [lower instance priority]
-instance ordered_semiring.pos_mul_strict_mono : zero_lt.pos_mul_strict_mono α :=
-⟨λ x a b h, mul_lt_mul_of_pos_left h x.prop⟩
+instance ordered_semiring.to_pos_mul_mono : zero_lt.pos_mul_mono α :=
+⟨λ x a b h, mul_le_mul_of_nonneg_left h x.2.le⟩
 
 @[priority 200] -- see Note [lower instance priority]
-instance ordered_semiring.mul_pos_strict_mono : zero_lt.mul_pos_strict_mono α :=
-⟨λ x a b h, mul_lt_mul_of_pos_right h x.prop⟩
+instance ordered_semiring.to_mul_pos_mono : zero_lt.mul_pos_mono α :=
+⟨λ x a b h, mul_le_mul_of_nonneg_right h x.2.le⟩
 
-section nontrivial
+lemma mul_le_mul (hac : a ≤ c) (hbd : b ≤ d) (hb : 0 ≤ b) (hc : 0 ≤ c) : a * b ≤ c * d :=
+(mul_le_mul_of_nonneg_right hac hb).trans $ mul_le_mul_of_nonneg_left hbd hc
 
-variables [nontrivial α]
+lemma mul_nonneg_le_one_le (h₁ : 0 ≤ c) (h₂ : a ≤ c) (h₃ : 0 ≤ b) (h₄ : b ≤ 1) : a * b ≤ c :=
```

Comment:
GitHub is having a hard time showing the diffs, did you just move these declarations around or did you change their hypotheses?

Reply:
Just moved them around.